### PR TITLE
Add MoE for row % and population

### PR DESCRIPTION
**Pull request by slobodan-ilic**
* drive by TDD (add failing tests)
* implement functionality as separate properties

### CODE REVIEW

File: src/cr/cube/cubepart.py [hunk 1]
```diff
@@ -298,15 +298,15 @@ def columns_margin(self):
...
 
     @lazyproperty
-    def columns_percentages_moe(self):
-        """1D/2D np.float64 ndarray of margin-of-error (MoE) for columns percentages.
+    def columns_proportions_moe(self):
+        """1D/2D np.float64 ndarray of margin-of-error (MoE) for columns proportions.
 
-        The values are represented as percentages, analogue to the `table_percentages`
-        property. This means that the value of 3.5% will have the value 3.5 (not 0.035).
+        The values are represented as fractions, analogue to the `table_proportions`
```
Comment (ernestoarbitrio) [inline src/cr/cube/cubepart.py:304]: `s/table_proportions/column_proportions`

File: src/cr/cube/cubepart.py [hunk 2]
```diff
@@ -459,6 +459,19 @@ def population_counts(self):
             self.table_proportions * self._population * self._cube.population_fraction
         )
 
+    @lazyproperty
+    def population_moe(self):
+        """2D np.float64 ndarray of population margin-of-error (MoE) for table percents.
+
+        The values are represented as population estimates, analogue to the
+        `population_counts` property. This means that the values will be presented by
+        actual estimated counts of the population The values can be np.nan when the
```
Comment (ernestoarbitrio) [inline src/cr/cube/cubepart.py:468]: `.` missing after population

File: src/cr/cube/cubepart.py [hunk 3]
```diff
@@ -526,6 +539,22 @@ def rows_dimension_type(self):
...
+        `row_proportions` property. This means that the value of 3.5% will have the
+        value 0.035. The values can be np.nan when the corresponding percentage is also
+        np.nan, which happens when the respective table margin is 0.
+        """
+        return Z_975 * self.rows_std_err
+
+    @lazyproperty
+    def rows_std_err(self):
+        """2D np.float64 ndarray of standard errors for row percentages """
+        return np.sqrt(self._rows_variance / self.rows_margin[:, None])
```
Comment (ernestoarbitrio) [inline src/cr/cube/cubepart.py:556]: Question, have we planned to expose the `rows_std_err` ? In the previous PRD we only had the col_std_dev and col_std_err. If yes maybe we need the `rows_std_dev` too.

File: tests/integration/test_multiple_response.py [hunk 4]
```diff
@@ -263,173 +264,29 @@ def test_table_base_unpruned_cat_x_mr():
...
+        slice_.columns_std_err, load_python_expression("mr-x-cat-weighted-col-std-err")
+    )
+    np.testing.assert_almost_equal(
+        slice_.columns_proportions_moe,
+        load_python_expression("mr-x-cat-weighted-col-moe"),
     )
     np.testing.assert_almost_equal(
-        slice_.table_percentages_moe, expected_table_percentages_moe
+        slice_.table_proportions_moe,
+        load_python_expression("mr-x-cat-weighted-table-moe"),
```
Comment (ernestoarbitrio) [inline tests/integration/test_multiple_response.py:289]: nice grooming 👍🏽

**Review by ernestoarbitrio — CHANGES_REQUESTED** [review]
couple of comments :D

File: src/cr/cube/cubepart.py [hunk 5]
```diff
@@ -526,6 +539,27 @@ def rows_dimension_type(self):
...
+    @lazyproperty
+    def rows_std_err(self):
+        """2D np.float64 ndarray of standard errors for row percentages."""
+        return np.sqrt(self._rows_variance / self.rows_margin[:, None])
+
+    @lazyproperty
+    def rows_std_dev(self):
+        """2D np.float64 ndarray of standard deviation for row percentages."""
+        return np.sqrt(self._rows_variance)
+
```
Comment (ernestoarbitrio) [inline src/cr/cube/cubepart.py:562]: please swap this 2 prop to respect the alphabetical order

File: src/cr/cube/cubepart.py [hunk 6]
```diff
@@ -526,6 +539,27 @@ def rows_dimension_type(self):
...
+    @lazyproperty
+    def row_std_err(self):
+        """2D np.float64 ndarray of standard errors for row percentages."""
+        return np.sqrt(self._row_variance / self.rows_margin[:, None])
+
+    @lazyproperty
+    def row_std_dev(self):
+        """2D np.float64 ndarray of standard deviation for row percentages."""
+        return np.sqrt(self._row_variance)
+
```
Comment (ernestoarbitrio) [inline src/cr/cube/cubepart.py:562]: this should be swapped to respect the alphabetical oder

File: src/cr/cube/cubepart.py [hunk 7]
```diff
@@ -900,6 +934,12 @@ def _columns_variance(self):
             self.counts / self.columns_margin * (1 - self.counts / self.columns_margin)
         )
 
+    @lazyproperty
+    def _row_variance(self):
+        """ndarray of variances for row percentages"""
+        margin = self.rows_margin[:, None]
```
Comment (ernestoarbitrio) [inline src/cr/cube/cubepart.py:940]: I don't understand why are you adding an extra dimension here? Can you explain please?
Furthermore I'd prefer to be more explicit here IMHO: `self._rows_margin[:, np.newaxis]` spells out better than using `None`. This way we can read better that we are adding an extra dimension

Comment (ernestoarbitrio) [reply src/cr/cube/cubepart.py:940]: Ah this property is not placed in its alphabetical right place. Please order it 😄

**Review by ernestoarbitrio — CHANGES_REQUESTED** [review]
left couple of comments 😄

File: src/cr/cube/cubepart.py (hunk 7 above)
Comment (slobodan-ilic) [reply src/cr/cube/cubepart.py]: Added comment

**Review by ernestoarbitrio — APPROVED** [review]
LGTM :)

**Pull request merged**